app/src/main/java/org/xwiki/android/sync/activities/OIDC/OIDCActivity.kt

```diff
@@ -30,36 +31,29 @@ import org.xwiki.android.sync.utils.AccountClickListener
 import org.xwiki.android.sync.utils.extensions.TAG
 import java.io.IOException
 import android.webkit.CookieSyncManager
+import okhttp3.*
+import org.json.JSONObject
 import org.xwiki.android.sync.utils.OIDCWebViewClient
 import org.xwiki.android.sync.utils.WebViewPageLoadedListener
+import java.net.URL
 
-private suspend fun createAuthorizationCodeFlow(selectedAccountName: String, serverUrl: String): AuthorizationCodeFlow {
-    val userServerBaseUrl: String = if (serverUrl.isNullOrEmpty()) {
-        userAccountsRepo.findByAccountName(
-            selectedAccountName
-        ) ?.serverAddress ?: throw IllegalArgumentException(
-            "Selected account name is absent in databse"
-        )
-    } else {
-        serverUrl
-    }
-
+private suspend fun createAuthorizationCodeFlow(serverUrl: String): AuthorizationCodeFlow {
     return AuthorizationCodeFlow.Builder(
         BearerToken.authorizationHeaderAccessMethod(),
         NetHttpTransport(),
         JacksonFactory(),
         GenericUrl(
             buildOIDCTokenServerUrl(
-                userServerBaseUrl
+                serverUrl
             )
         ),
         ClientParametersAuthentication(
-            selectedAccountName,
+            Settings.Secure.ANDROID_ID,
             ""
         ),
-        selectedAccountName,
+        Settings.Secure.ANDROID_ID,
```

Comment:
Why this was replaced?

Reply:
@tmortagne  Said that "There's no need to send username/selectedAccount, that not how it works. Simply send a unique Id instead of the username" That's why I'm sending this unique ID. "Settings.Secure.ANDROID_ID"

Reply:
ok, I see